core/include/vecmem/utils/deleter.ipp

```diff
@@ -0,0 +1,24 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+namespace vecmem::details {
+
+   template< typename T, typename Allocator >
+   deleter< T, Allocator >::
+   deleter( size_type elements, const allocator_type& allocator )
+   : m_elements( elements ), m_allocator( allocator ) {
+
+   }
+
+   template< typename T, typename Allocator >
+   void deleter< T, Allocator >::operator()( void* ptr ) {
+
+      m_allocator.deallocate( static_cast< T* >( ptr ), m_elements );
```

Comment:
This still doesn't work, I think. `std::allocator<T>::deallocate` states that it calls `::operator delete(void*, std::align_val_t)`, which is a [class-agnostic](https://en.cppreference.com/w/cpp/memory/new/operator_delete) variant of operator delete which does not delete.

Reply:
I'm afraid not. The [documentation for `std::pmr::polymorphic_allocator<T>::deallocate`](https://en.cppreference.com/w/cpp/memory/polymorphic_allocator/deallocate) states that it is equivalent to `this->resource()->deallocate(p, n * sizeof(T), alignof(T));`. That doesn't do any destruction. What you want is [`std::pmr::polymorphic_allocator<T>::delete_object`](https://en.cppreference.com/w/cpp/memory/polymorphic_allocator/delete_object), but that's a C++20 feature... :disappointed: